mParticle-Apple-SDK/Next Gen/MPEventLogging.m

```diff
@@ -0,0 +1,353 @@
+//
+//  MPEventLogging.m
+//  mParticle-Apple-SDK
+//
+//  Created by Ben Baron on 6/29/23.
+//
+
+#import "MPEventLogging.h"
+#import "MPBackendController.h"
+#import "MPAppDelegateProxy.h"
+#import "MPPersistenceController.h"
+#import "MPMessage.h"
+#import "MPSession.h"
+#import "MPIConstants.h"
+#import "MPStateMachine.h"
+#import "MPNetworkPerformance.h"
+#import "MPIUserDefaults.h"
+#import "MPBreadcrumb.h"
+#import "MPUpload.h"
+#import "MPApplication.h"
+#import "MPCustomModule.h"
+#import "MPMessageBuilder.h"
+#import "MPEvent.h"
+#import "MParticleUserNotification.h"
+#import "NSDictionary+MPCaseInsensitive.h"
+//#import "MPHasher.h"
+#import "MPUploadBuilder.h"
+#import "MPILogger.h"
+#import "MPResponseEvents.h"
+#import "MPConsumerInfo.h"
+#import "MPResponseConfig.h"
+#import "MPCommerceEvent.h"
+#import "MPCommerceEvent+Dictionary.h"
+#import "MPKitContainer.h"
+#import "MPUserAttributeChange.h"
+#import "MPUserIdentityChange.h"
+#import "MPSearchAdsAttribution.h"
+#import "MPURLRequestBuilder.h"
+#import "MPArchivist.h"
+#import "MPListenerController.h"
+#import "MParticleWebView.h"
+#import "MPDevice.h"
+
+@interface MParticle()
+@property (nonatomic, strong, nonnull) MPBackendController *backendController;
+@property (nonatomic, strong) MPStateMachine *stateMachine;
+@property (nonatomic, strong) MPPersistenceController *persistenceController;
+@end
+
+@implementation MPEventLogging
```

Comment:
The implementation itself is kind of a mixture between an adapter (old/new interfaces and architecture) and a delegate right?

If we want to iterate over this in time to create extra layer to achieve high cohesivness and almost single responsibility of the implementation classes we might want to create an adapter and make this call the adapter class which will actually have the business logic.
That way it might be easier to iterate (just an idea)

logEvent() {
     //add params based on new interface implementation and pass it to the adapter function. 
     adapter.logEvent()
 }

On top of it.. the MPEventLogging should be acting almost as a visitor in the registered kits? 
The two prev concepts match if we were implement this as:

logEvent() {
    mediator.getKits.forEach {. kit ->
         kit.logEvent() // by doing this each kit will have a concrete impl and the mPKit have an adapter internally
    }
}

Reply:
MPEventLogging at this point just groups the existing event logging methods so they can be refactored and shimmed/adapted later. I really wanted to reduce the scope of the initial architecture changes as much as possible.